pkg/controller/core/workload_controller.go

```diff
@@ -467,9 +468,15 @@ func (r *WorkloadReconciler) triggerDeactivationOrBackoffRequeue(ctx context.Con
 		Steps:    int(requeuingCount),
 	}
 	var waitDuration time.Duration
+	maxWaitDuration := time.Duration(r.waitForPodsReady.requeuingBackoffMaxSeconds) * time.Second
 	for backoff.Steps > 0 {
-		waitDuration = backoff.Step()
+		newWaitDuration := backoff.Step()
+		if newWaitDuration > maxWaitDuration {
+			break
+		}
+		waitDuration = newWaitDuration
```

Comment:
```suggestion
		waitDuration := min(backoff.Step(), maxWaitDuration)
```
seems much easier to read